slack-clone-compose-sample/src/main/kotlin/io/getstream/slack/compose/ui/channels/ChannelItem.kt

```diff
@@ -0,0 +1,197 @@
+import io.getstream.chat.android.compose.ui.common.avatar.UserAvatar
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+import io.getstream.chat.android.compose.ui.util.getDisplayName
+import io.getstream.slack.compose.R
+import io.getstream.slack.compose.ui.channels.components.OnlineIndicator
+import io.getstream.slack.compose.ui.channels.components.UnreadCountBadge
+import io.getstream.slack.compose.ui.util.getOtherUser
+
+/**
+ * Component that represents a one-to-one channel item.
+ *
+ * One-to-one channel is a distinct channel with only two members.
+ *
+ * @param channel The channel to display.
+ * @param onChannelClick Handler for a single tap on an item.
+ * @param modifier Modifier for styling.
+ */
+@Composable
+fun DirectOneToOneChatItem(
+    channel: Channel,
+    onChannelClick: (Channel) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Row(
+        modifier = modifier
+            .clip(shape = RoundedCornerShape(6.dp))
+            .clickable(onClick = { onChannelClick(channel) })
+            .padding(horizontal = 8.dp)
+            .fillMaxSize(),
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        val user = channel.getOtherUser()!!
+
+        Box(
+            modifier = Modifier
+                .clip(ChatTheme.shapes.avatar)
+                .height(32.dp)
+                .width(30.dp),
+        ) {
+            UserAvatar(
+                modifier = Modifier
+                    .width(24.dp)
+                    .height(24.dp)
+                    .align(Alignment.CenterStart),
+                user = user
+            )
+            OnlineIndicator(
+                modifier = Modifier
+                    .align(Alignment.BottomEnd)
+                    .size(14.dp),
+                online = user.online
+            )
+        }
+
+        Spacer(Modifier.width(6.dp))
+        ChannelName(
+            channel = channel,
+            modifier = Modifier.weight(1f),
+        )
+
+        val unreadCount = channel.unreadCount ?: 0
+        if (unreadCount > 0) {
+            UnreadCountBadge(unreadCount)
+        }
+    }
+}
+
+/**
+ * Component that represents a distinct channel item.
+ *
+ * A distinct channel is a channel created without ID based on members.
+ *
+ * @param channel The channel to display.
+ * @param onChannelClick Handler for a single tap on an item.
+ * @param modifier Modifier for styling.
+ */
+@Composable
+fun DirectGroupChatItem(
+    channel: Channel,
+    onChannelClick: (Channel) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Row(
+        modifier = modifier
+            .clip(shape = RoundedCornerShape(6.dp))
+            .clickable(onClick = { onChannelClick(channel) })
+            .padding(horizontal = 8.dp)
+            .fillMaxSize(),
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        Box(
+            modifier = modifier
+                .clip(RoundedCornerShape(4.dp))
```

Comment:
You can try using the modifiers directly on the Text so that we don't have an extra parent here.

Also for the Text, we can use number.toString() instead of concatenation.

We could also try using a Typography from the theme for the type that makes sense (e.g. maybe just body, or body bold)